pkg/pool-weighted/contracts/WeightedPoolProtocolFees.sol

```diff
@@ -45,7 +45,7 @@ abstract contract WeightedPoolProtocolFees is BaseWeightedPool, ProtocolFeeCache
     // immutable, the invariant only changes due to accumulated swap fees, which saves gas by freeing the Pool
     // from performing any computation or accounting associated with protocol fees during swaps.
     // This mechanism requires keeping track of the invariant after the last join or exit.
-    uint256 private _lastPostJoinExitInvariant;
+    uint256 internal _lastPostJoinExitInvariant;
```

Comment:
I'd really like to keep this private. There is a public getter `getLastInvariant()`, so what you can do is leave this private and call that public getter from the Mock.

Could do the same thing for the yield: make it private, add a public getter, and call that in the Mock.

It's a little dangerous / non-future-proofed to make mutable important things non-private, when the intent is to encapsulate their behavior in this protocol fees contract.

Reply:
Ah, I missed that getter, I swtich to use that. One downside to this however is that we're forced into spending the bytecode in the production contract for a getter which is only relevant for testing.

Reply:
I guess we have the bytecode to spare for now so no harm.

Reply:
Is it only relevant for testing? If so, the getter could be internal (getting a little back).

It seems like the ATH Rate might be useful for UIs, etc. That getter could be public